### ISSUE TITLE
Keyboard navigation: when snapToGrid is turned on, the step size should be snapSizeX/Y

### ISSUE THREAD

**Issue by christianp** [jsxgraph/jsxgraph#380]
When moving a point with the keyboard, if `snapToGrid = true` and the step size is less than half of the width of the grid, then it's impossible to move the point - its position always gets rounded to the same grid point.

So if `snapToGrid` is turned on, I think `snapSizeX` should override `keyboard.dx` (and similarly for `y`).

**Comment by christianp** [comment]
Thanks!

**Comment by christianp** [comment]
I think that this still isn't quite right. Here's an example: https://codepen.io/christianp/pen/abXrJMo

When the automatically calculated keyboard distance is bigger than `snapSizeX/Y`, the point moves the bigger distance. I think it should always move by `snapSizeX/Y`, when explicitly set. Is there a way to tell if it was explicitly set?

**Comment by alfredwassermann** [comment]
Thank you very much for pointing this out. 

Since `snapToGrid:true`  means the point jumps from grid point to grid point,  I simply set the step width of the keyboard navigation to the distance between two grid points, see commit fd7e64ba59166ee9b1b620faba5c2fee7d7a9093
